ledger/apply/stateproof.go

```diff
@@ -41,31 +42,54 @@ func StateProof(tx transactions.StateProofTxnFields, atRound basics.Round, sp St
 	}
 
 	lastRoundInInterval := basics.Round(tx.Message.LastAttestedRound)
-	lastRoundHdr, err := sp.BlockHdr(lastRoundInInterval)
-	if err != nil {
-		return err
-	}
-
 	nextStateProofRnd := sp.GetStateProofNextRound()
 	if nextStateProofRnd == 0 || nextStateProofRnd != lastRoundInInterval {
 		return fmt.Errorf("applyStateProof: %w - expecting state proof for %d, but new state proof is for %d",
 			ErrExpectedDifferentStateProofRound, nextStateProofRnd, lastRoundInInterval)
 	}
 
-	proto := config.Consensus[lastRoundHdr.CurrentProtocol]
-	if validate {
-		votersRnd := lastRoundInInterval.SubSaturate(basics.Round(proto.StateProofInterval))
-		votersHdr, err := sp.BlockHdr(votersRnd)
-		if err != nil {
-			return err
-		}
+	atRoundHdr, err := sp.BlockHdr(atRound)
+	if err != nil {
+		return err
+	}
+
+	var verificationData *ledgercore.StateProofVerificationData
+	if config.Consensus[atRoundHdr.CurrentProtocol].StateProofUseTrackerVerification {
```

Comment:
Do you think we should cover the verification method switch with a test?

Reply:
There are two tests - 
1 TestCowStateProof check the old code - verification using blockheader
2 TestCowStateProofV34 - using the tracker.